### PR TITLE
Ignore attestations with subject mismatch

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-1298

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

**Review by zregvart — APPROVED** [review]
Cool

File: internal/evaluation_target/application_snapshot_image/application_snapshot_image.go [hunk 1]
```diff
@@ -246,6 +246,39 @@ func (a ApplicationSnapshotImage) ValidateAttestationSyntax(ctx context.Context)
 	return EV003.CausedBy(errors.New(msg))
 }
 
+// FilterMatchingAttestations ignores attestations that do not have a matching subject.
+func (a *ApplicationSnapshotImage) FilterMatchingAttestations(ctx context.Context) {
+	filteredAttestations := make([]oci.Signature, 0, len(a.attestations))
+	for _, attestation := range a.attestations {
+		_, statement, err := statementFrom(ctx, attestation)
+		if err != nil {
+			log.Debugf("Unable to extract statement from attestation: %v", err)
```
Comment (lcarva) [inline internal/evaluation_target/application_snapshot_image/application_snapshot_image.go:255]: Oh shoot. I just realized there should be a `continue` here. I'll see if I can add a unit test to capture this case.

Comment (lcarva) [reply internal/evaluation_target/application_snapshot_image/application_snapshot_image.go:255]: Done.

**Review by zregvart — APPROVED** [review]
Cool, missed the continue, looked very straightforward to me, nice catch.

**Pull request merged**